tools/ci/production-upgrade

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+# This test is for upgrading the production server
+set -e
+set -x
+
+# Uninstall rabbitmq-server and supervisor as this has configurations generated for previous containers
+# We install it again from scratch.
+sudo apt-get remove rabbitmq-server supervisor && sudo apt-get purge rabbitmq-server supervisor
+sudo apt-get install rabbitmq-server supervisor
+
+# Start the postgresql service.
+sudo service postgresql start
+
+# Starting the rabbitmq-server
+if ! sudo service rabbitmq-server start; then
+    echo
+    echo "Starting rabbitmq-server failed. Trying again:"
+    sudo service rabbitmq-server start
+fi
+
+# Applying puppet changes. This also regenerates the supervisor configurations.
+sudo /home/zulip/deployments/current/scripts/zulip-puppet-apply -f
+
+# Stopping nginx service started by above command.
+# This causes unexpected `Unable to stop Service[nginx]` error in the puppet apply step of upgrade otherwise.
+if ! sudo service nginx stop; then
+    echo
+    echo "Stoping nginx failed. Trying again:"
+    sudo service nginx stop
+fi
+
+# Perform the upgrade.
+sudo /home/zulip/deployments/current/scripts/upgrade-zulip /tmp/zulip-server-test.tar.gz && {
+    ret=$?
+    if [ "$ret" -eq 0 ]; then
+        # We don't have the production-verify step currently due to some container issues.
+        # So to check that the upgrade went successfully, we check the value of version
+        # file. For the current version, its value evaluates to the version value. After upgrading,
+        # its value changes to "test". Since this change occurs in the last steps just before restarting
+        # the server, it is the safest to perform. Note that this is required since
+        # the upgrade script in the previous releases exits out with a zero status even
+        # if it fails to upgrade.
+        if [ "$(cat /home/zulip/deployments/current/version)" = "test" ]; then
+            echo
+            echo "Upgrade of production server complete!"
+            exit 0
+        fi
+    fi
+}
```

Comment:
I think we can replace this hackery with just calling `scripts/lib/upgrade-zulip`; I'm going to try that in a push back here, and if it passes, we can merge.